File: sdkwork-ai-sdk-swift/Sources/API/ContextApi.swift
import Foundation

public class ContextApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Chat with context
    public func chatWith(context_id: String, body: ChatWithContextRequest) async throws -> ContextResponse? {
        let response = try await client.post(ApiPaths.aiPath("/context/\(context_id)/chat/completions"), body: body)
        return response as? ContextResponse
    }

    /// Create context
    public func createContext(body: ContextCreateRequest) async throws -> ContextResponse? {
        let response = try await client.post(ApiPaths.aiPath("/context/create"), body: body)
        return response as? ContextResponse
    }

    /// Get context
    public func getContext(context_id: String) async throws -> ContextResponse? {
        let response = try await client.get(ApiPaths.aiPath("/context/\(context_id)"))
        return response as? ContextResponse
    }

    /// Delete context
    public func deleteContext(context_id: String) async throws -> Void {
        _ = try await client.delete(ApiPaths.aiPath("/context/\(context_id)"))
    }
}

File: sdkwork-ai-sdk-csharp/Api/ContextApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class ContextApi
    {
        private readonly HttpClient _client;

        public ContextApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Chat with context
        /// </summary>
        public async Task<ContextResponse?> ChatWithAsync(string context_id, ChatWithContextRequest body)
        {
            return await _client.PostAsync<ContextResponse>(ApiPaths.AiPath($"/context/{context_id}/chat/completions"), body);
        }

        /// <summary>
        /// Create context
        /// </summary>
        public async Task<ContextResponse?> CreateContextAsync(ContextCreateRequest body)
        {
            return await _client.PostAsync<ContextResponse>(ApiPaths.AiPath("/context/create"), body);
        }

        /// <summary>
        /// Get context
        /// </summary>
        public async Task<ContextResponse?> GetContextAsync(string context_id)
        {
            return await _client.GetAsync<ContextResponse>(ApiPaths.AiPath($"/context/{context_id}"));
        }

        /// <summary>
        /// Delete context
        /// </summary>
        public async Task DeleteContextAsync(string context_id)
        {
            await _client.DeleteAsync<object>(ApiPaths.AiPath($"/context/{context_id}"));
        }
    }
}
